src/argilla/training/base.py

```diff
@@ -0,0 +1,163 @@
+        )
+        if framework in ["transformers", "setfit"]:
+            import torch
+
+            if torch.backends.mps.is_available():
+                self.device = "mps"
+            elif torch.cuda.is_available():
+                self.device = "cuda"
+            else:
+                self.device = "cpu"
+
+        if framework == "setfit":
+            assert self._rg_dataset_type == rg.DatasetForTextClassification, "SetFit supports only text classification"
+            self._trainer = ArgillaSetFitTrainer(
+                record_class=self._rg_dataset_type._RECORD_TYPE,
+                dataset=self.dataset_full_prepared,
+                multi_label=self._multi_label,
+                device=self.device,
+                seed=self._seed,
+            )
+        else:
+            raise NotImplementedError(f"Framework {framework} is not supported")
+        self._logger.warning(self)
+
+    def __repr__(self) -> str:
+        """
+        `trainer.__repr__()` prints out the trainer's parameters and a summary of how to use the trainer
+
+        Returns:
+          The trainer object.
+        """
+        return inspect.cleandoc(
+            f"""
+            ArgillaBaseTrainer info:
+            _________________________________________________________________
+            These baseline params are fixed:
+                dataset: {self._name}
+                task: {self._rg_dataset_type.__name__}
+                multi_label: {self._multi_label}
+                required_fields: {self._required_fields}
+                train_size: {self._train_size}
+
+            {self._trainer.__class__} info:
+            _________________________________________________________________
+            The parameters are configurable via `trainer.update_config()`:
+                {self._trainer}
+
+            Using the trainer:
+            _________________________________________________________________
+            `trainer.train(path)` to train to start training. `path` is the path to save the model automatically.
+            `trainer.predict(text, as_argilla_records=True)` to make predictions.
+            `trainer.save(path)` to save the model manually."""
+        )
+
+    def update_config(self, *args, **kwargs):
+        """
+        It updates the configuration of the trainer, but the parameters depend on the trainer.subclass.
+        """
+        self._trainer.update_config(*args, **kwargs)
+
+    def predict(self, text: Union[List[str], str], as_argilla_records: bool = True):
+        """
+        `predict` takes a string or list of strings and returns a list of dictionaries, each dictionary
+        containing the text, the predicted label, and the confidence score
+
+        Args:
+          text (Union[List[str], str]): The text to be classified.
+          as_argilla_records (bool): If True, the output will be a list of ArgillaRecord objects. If
+        True, the output will be a list of Argilla records. Defaults to True.
```

Comment:
```suggestion
          as_argilla_records (bool): If True, the output will be a list of Argilla records instead of dictionaries. Defaults to True.
```
Duplicated, and my update mentions the dictionaries, too.